src/form-elements/FormElementGoogleAddress.tsx

```diff
@@ -0,0 +1,196 @@
+            {
+              input,
+            },
+            (predictions, status) => {
+              if (status !== google.maps.places.PlacesServiceStatus.OK) {
+                reject('Google Places service not available')
+                return
+              }
+              resolve(predictions ?? [])
+              return
+            },
+          )
+        }).catch((e) => {
+          if (!abortSignal.aborted) {
+            Sentry.captureException(e)
+          }
+          throw new OneBlinkAppsError(
+            'An unknown error has occurred. Please contact support if the problem persists.',
+            { originalError: e },
+          )
+        })
+
+        if (!abortSignal.aborted) {
+          return predictions.reduce<Array<{ value: string; label: string }>>(
+            (results, prediction) => {
+              if (prediction.place_id) {
+                results.push({
+                  value: prediction.place_id,
+                  label: prediction.description,
+                })
+              }
+              return results
+            },
+            [],
+          )
+        }
+      }
+      throw new OneBlinkAppsError(
+        'An unknown error has occurred. Please contact support if the problem persists.',
+      )
+    },
+    [autocompleteService],
+  )
+
+  const handleChange = React.useCallback(
+    async (placeId: string | undefined) => {
+      if (!placeId) {
+        onChange(element, { value: undefined })
+        return
+      }
+
+      setIsLoadingAddressDetails(true)
+      try {
+        //this should not happen, we can't get a place Id without google being present
+        if (!isLoaded) {
+          throw new OneBlinkAppsError(
+            'An unknown error has occurred. Please contact support if the problem persists.',
+            {
+              originalError: new Error(
+                'google places library has not be initialised',
+              ),
+            },
+          )
+        }
+
+        const placeService = new google.maps.places.Place({ id: placeId })
+        const { place } = await placeService.fetchFields({
+          fields: [
+            'id',
+            'displayName',
+            'formattedAddress',
+            'location',
+            'addressComponents',
+            'servesBeer',
+          ],
+        })
+        onChange(element, { value: place })
+      } catch (newError) {
+        if (isMounted.current) {
+          setError(newError as Error)
+        }
+      }
+      if (isMounted.current) {
+        setIsLoadingAddressDetails(false)
+      }
+    },
+    [isMounted, onChange, element, isLoaded],
+  )
+
+  // Ensure the label is set if the value is set outside of this component
+  React.useEffect(() => {
+    if (value) {
+      const newLabel = value.formattedAddress || value.id
+      if (label !== newLabel) {
+        setLabel(newLabel || '')
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Prob want to whack a comment specifically before/on this line to say what we need it.